src/Validator/Parser/hdtXSDSchemaParser.cpp

```
@@ -0,0 +1,413 @@
+					if (inCompositor)
+						continue;
+					collectTypeConstraints(child, inCompositor, elemName, namedSimpleTypes, textCons, attrCons);
+				} else {
+					collectTypeConstraints(child, inCompositor, elemName, namedSimpleTypes, textCons, attrCons);
+				}
+			}
+		}
+
+		/// Recursively gathers typed attributes declared inside a complexType node.
+		void gatherComplexTypeAttrs(
+			pugi::xml_node n,
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'gatherComplexTypeAttrs' of similar type are easily swapped by mistake

(bugprone-easily-swappable-parameters)